### ISSUE TITLE
Request: Add FiveM

### ISSUE THREAD

**Issue by Kreyren** [PhoenicisOrg/scripts#914]
![image](https://user-images.githubusercontent.com/11302521/56455259-02fc4280-6354-11e9-89ab-373de0fb16fe.png)

FiveM (https://fivem.net/) is open-source Grand Theft Auto V multiplayer framework for singleplayer.

Relevant: https://github.com/PhoenicisOrg/phoenicis/issues/1919

TODO: Add AppDB entry with test-info.

--- 

Test-Cases wanted!!

**Comment by plata** [comment]
Can you try if simply using staging is sufficient to make this work?

**Comment by Kreyren** [comment]
@plata using quick script with wine-staging is good start and should work based on all provided info.

test cases needed can't test atm, is in my TODO.

Would recommend pushing quick script.

**Comment by plata** [comment]
I do not own GTA V, so I cannot test this myself.

**Comment by Kreyren** [comment]
> I do not own GTA V, so I cannot test this myself.

I sent you the key on stack, go enjoy.

**Comment by suomilanittaja** [comment]
any news?

**Comment by plata** [comment]
According to @Kreyren, `OnlineInstallerScript` with Wine staging should work. Can you try that?

**Comment by suomilanittaja** [comment]
stuck this place using amd64 wine 4.6 staging  https://ibb.co/kB3LvPB

**Comment by suomilanittaja** [comment]
> According to @Kreyren, `OnlineInstallerScript` with Wine staging should work. Can you try that?

i tested wine 4.6 staging (amd64) i get tihs error https://ibb.co/tMkPd8N when i press ok i get other problem which cant be ignored https://ibb.co/tBz8K6X

**Comment by plata** [comment]
Can you try with Windows 10 like suggested there?

**Comment by suomilanittaja** [comment]
i have configured wine to use windows 10

**Comment by plata** [comment]
Mh. In https://bugs.launchpad.net/ubuntu/+source/playonlinux/+bug/1750155 it is suggested to run the exe like
```
wine runas /trustlevel:0x20000 xyz.exe
```
Does this help?

**Comment by suomilanittaja** [comment]
Yes now i can go forward lets wait next problem

**Comment by plata** [comment]
Good. I don't think it's currently possible to do this in a nice way in a quick script. We might have to think about this. Maybe something like `.trustlevel("0x20000")`. Plugin would be nicer but I don't see how this could work currently. Ideas welcome.

**Comment by suomilanittaja** [comment]
there next problem cant write my account https://ibb.co/J23C0Lh

**Comment by plata** [comment]
Any error message or similar issue reported on Wine bugzilla?

**Comment by suomilanittaja** [comment]
i try to install steam windows version to skip rockstar studio account asking

**Comment by plata** [comment]
Ok, by the way: you can embed screenshots directly in your post on GitHub.

**Comment by Kreyren** [comment]
info: i started working on this.

**Comment by suomilanittaja** [comment]
okey it cant be skipped so it just need to get work

**Comment by Kreyren** [comment]
> okey it cant be skipped so it just need to get work
> maybe this help https://gist.github.com/BeauBouchard/c496fcd231c3d06fced5

afaik outdated those should be staged in wine-staging with workaround for dxvk, dotnet46+ for mods and stuff and vcrun2017 (maybe). On my setup i had to only install dxvk. 

55% downloading GTA V so far FiveM works .

**Comment by suomilanittaja** [comment]
i installed corefonts and now i able to write my account but there is new problem ![Screenshot from 2019-04-27 18-21-17](https://user-images.githubusercontent.com/29737152/56851585-7c2e0380-6919-11e9-9dc8-00666f75fa14.png) and console look like this 
![Screenshot from 2019-04-27 18-21-35](https://user-images.githubusercontent.com/29737152/56851598-9a93ff00-6919-11e9-8ca5-360bdb3ad37f.png)

**Comment by Kreyren** [comment]
@suomilanittaja Provide output of following:

`wine --version`

`winetricks --version`

`env WINEPREFIX="<path_to_GTAV_WINEPREFIX> winetricks list-installed`

is GTA V even working on your system?; else this is not relevant.

**Comment by suomilanittaja** [comment]
![Screenshot from 2019-04-27 18-41-04](https://user-images.githubusercontent.com/29737152/56851814-07a89400-691c-11e9-9f4c-756e8ab5f014.png)
i dont have winetricks but i have installed thinks in pol5

**Comment by Kreyren** [comment]
## ROADBLOCK 01

```console
kreyren@dreamon ~/Games/FiveM $ info:  Game: FiveM.exe
info:  DXVK: v1.0.3
warn:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
WARNING: radv is not a conformant vulkan implementation, testing use only.
info:  AMD RADV PITCAIRN (LLVM 8.0.0):
info:    Driver: 19.0.0
info:    Vulkan: 1.1.90
info:    Memory Heap[0]:
info:      Size: 1792 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0x1
info:    Memory Heap[1]:
info:      Size: 256 MiB
info:      Flags: 0x1
info:      Memory Type[2]: Property Flags = 0x7
info:    Memory Heap[2]:
info:      Size: 3072 MiB
info:      Flags: 0x0
info:      Memory Type[1]: Property Flags = 0x6
info:      Memory Type[3]: Property Flags = 0xe
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
info:  Enabled device extensions:
info:    VK_EXT_memory_priority
info:    VK_EXT_shader_viewport_index_layer
info:    VK_EXT_transform_feedback
info:    VK_EXT_vertex_attribute_divisor
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_driver_properties
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
warn:  DXVK: No state cache file found
warn:  DXVK: Creating new state cache file
info:  DXVK: Using 4 compiler threads
warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
```

![image](https://user-images.githubusercontent.com/11302521/56851961-daa0b500-690c-11e9-9963-5265e810cbbe.png)

![image](https://user-images.githubusercontent.com/11302521/56851965-ea1ffe00-690c-11e9-8943-0b79fbcfc032.png)

**Comment by Kreyren** [comment]
> ![Screenshot from 2019-04-27 18-41-04](https://user-images.githubusercontent.com/29737152/56851814-07a89400-691c-11e9-9f4c-756e8ab5f014.png)
> i dont have winetricks but i have installed thinks in pol5
> and gta v dont work with proton becouse i have too old drivers and newer drivers dont work with me gpu

then this is not helpful, you need dxvk to run the game based on my recent (2 mins old) test result. -> Install DXVK using phoenicis or get winetricks or export it manually.

**Comment by suomilanittaja** [comment]
@Kreyren use (wine runas /trustlevel:0x20000 FiveM.exe) to fix run as admistrator bug
and i have dxvk installed

**Comment by Kreyren** [comment]
@suomilanittaja that was helpful :D thanks

**Comment by Kreyren** [comment]
@suomilanittaja 
![image](https://user-images.githubusercontent.com/11302521/56852083-67983e00-690e-11e9-872e-b38a9d38b7fb.png)


....

**Comment by Kreyren** [comment]
@suomilanittaja 
> i have dxvk installed

wine-staging + dxvk is mandatory for GTA V on my end. -> that doesn't work for you?

**Comment by suomilanittaja** [comment]
no it dosent but get you fivem work for you

**Comment by Kreyren** [comment]
> no it dosent but get you fivem work for you

Assumig that we have GTA V script in phoenicis try that one, else file a new issue i will contrib it -> Do not resolve GTA V issues here.

**Comment by Kreyren** [comment]
## ROADBLOCK 02

![image](https://user-images.githubusercontent.com/11302521/56852331-f9557a80-6911-11e9-86b6-813a5639f4e0.png)

```console
kreyren@dreamon ~/Games/FiveM $ info:  Game: FiveM.exe
info:  DXVK: v1.0.3
warn:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
WARNING: radv is not a conformant vulkan implementation, testing use only.
info:  AMD RADV PITCAIRN (LLVM 8.0.0):
info:    Driver: 19.0.0
info:    Vulkan: 1.1.90
info:    Memory Heap[0]:
info:      Size: 1792 MiB
info:      Flags: 0x1
info:      Memory Type[0]: Property Flags = 0x1
info:    Memory Heap[1]:
info:      Size: 256 MiB
info:      Flags: 0x1
info:      Memory Type[2]: Property Flags = 0x7
info:    Memory Heap[2]:
info:      Size: 3072 MiB
info:      Flags: 0x0
info:      Memory Type[1]: Property Flags = 0x6
info:      Memory Type[3]: Property Flags = 0xe
info:  D3D11CoreCreateDevice: Probing D3D_FEATURE_LEVEL_11_0
info:  D3D11CoreCreateDevice: Using feature level D3D_FEATURE_LEVEL_11_0
info:  Enabled device extensions:
info:    VK_EXT_memory_priority
info:    VK_EXT_shader_viewport_index_layer
info:    VK_EXT_transform_feedback
info:    VK_EXT_vertex_attribute_divisor
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_driver_properties
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
info:  DXVK: Read 0 valid state cache entries
info:  DXVK: Using 4 compiler threads
warn:  DXGI: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
Initialized system mapping!
CitizenFX Steam child starting - command line: "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.exe"
Initialized system mapping!
CitizenFX Steam child starting - command line: "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.exe"
Initialized system mapping!
CitizenFX Steam child starting - command line: "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cache\subprocess\FiveM_SteamChild.exe" -steamparent:69
Initializing Steam parent.
Initializing Steam parent: Steam's running.
Initiahello from "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.exe"
Got ros:legit process - pid 84
Initialized system mapping!
hello from "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cache\subprocess\FiveM_ROSLauncher" ros:legit --parent_pid=69 "C:\Program Files (x86)\Steam\steamapps\common\Grand Theft Auto V\GTAVLauncher.exe"
[0427/182654.218:ERROR:network_change_notifier_win.cc(156)] WSALookupServiceBegin failed with: 0

DevTools listening on ws://127.0.0.1:13173/devtools/browser/52dd617f-eb85-48e3-9cef-b069cc05ef37
Initialized system mapping!
Initialized system mapping!
[0427/182656.814:WARNING:cert_verify_proc_win.cc(104)] Unknown error -2146762482 mapped to net::ERR_FAILED
CitizenFX Steam child starting - command line: "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cache\subprocess\FiveM_AuthBrowser" --type=gpu-process --field-trial-handle=888,3953399459949674519,15513370790403341262,131072 --no-sandbox --locales-dir-path="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\bin/cef/" --log-file="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cef.log" --resources-dir-path="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\bin/cef/" --lang=en-US --gpu-preferences=KAAAAAAAAACAAwAAAQAAAAAAAAAAAGAAAAAAAAEAAAAIAAAAAAAAACgAAAAEAAAAIAAAAAAAAAAoAAAAAAAAADAAAAAAAAAAOAAAAAAAAAAQAAAAAAAAAAAAAAAFAAAAEAAAAAAAAAAAAAAABgAAABAAAAAAAAAAAQAAAAUAAAAQAAAAAAAAAAEAAAAGAAAA --use-gl=swiftshader-webgl --log-file="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cef.log" --service-request-channel-token=12827939048111765714 --mojo-platform-channel-handle=892 /prefetch:2
CitizenFX Steam child starting - command line: "C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cache\subprocess\FiveM_AuthBrowser" --type=renderer --no-sandbox --log-file="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cef.log" --field-trial-handle=888,3953399459949674519,15513370790403341262,131072 --disable-gpu-compositing --service-pipe-token=13845069489143070446 --lang=en-US --locales-dir-path="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\bin/cef/" --log-file="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\cef.log" --resources-dir-path="C:\users\kreyren\Local Settings\Application Data\FiveM\FiveM.app\bin/cef/" --disable-extensions --disable-pdf-extension --device-scale-factor=1.5 --num-raster-threads=4 --enable-main-frame-before-activation --service-request-channel-token=13845069489143070446 --renderer-client-id=3 --mojo-platform-channel-handle=1156 /prefetch:1
[0427/182701.527:ERROR:viz_main_impl.cc(169)] Exiting GPU process due to errors during initialization
[0427/182701.838:WARNING:cert_verify_proc_win.cc(104)] Unknown error -2146762482 mapped to net::ERR_FAILED
```

**Comment by Kreyren** [comment]
TRIGGERED
![image](https://user-images.githubusercontent.com/11302521/56852732-b4801280-6916-11e9-9f87-
dd92fe71c892.png)

**Comment by Kreyren** [comment]
@plata requresting HELP-WANTED

**Comment by Kreyren** [comment]
FiveM was submitted in WineHQ with results from this issue.

**Comment by Kreyren** [comment]
Investigating + waiting for more info -> seems to be blocked by DXVK, if so maybe using wine's dlls to work around this issue to be used for FiveM only so that Grand Theft Auto V woudn't be affected by performance loss?

**Comment by Kreyren** [comment]
Blocked by DXVK confirmed https://github.com/doitsujin/dxvk/issues/1033#event-2304844874

**Comment by Kreyren** [comment]
ping froggy @Tk-Glitch 

Any idea how to work around this issue? 

thanks for any info <3 

TLDR: DXVK doesn't support shared resources which seems to be required by FiveM (https://fivem.net/).

**Comment by plata** [comment]
If this is an issue in DXVK, I don't think there's anything we could do about it. You should update WineHQ app DB with your findings.
Could [OneSync](https://forum.fivem.net/t/the-onesync-eap-and-you/165931) be an alternative?

**Comment by suomilanittaja** [comment]
there are maybe two alternatives 1. make own fivem modification that runs with wine 2. try get mods like fivem work with pol5 like:https://gt-mp.net/ or https://rage.mp/

**Comment by Kreyren** [comment]
@plata 
> If this is an issue in DXVK, I don't think there's anything we could do about it. 
Investigating

>You should update WineHQ app DB with your findings.
Updated

> Could OneSync be an alternative?
elaborate?!

> there are maybe two alternatives 1. make own fivem modification that runs with wine
Unlikely to be worth the efford assuming it would be fork that is not considered as official and you would have to support it or rewrite fiveM

> . try get mods like fivem work with pol5 like:https://gt-mp.net/ or https://rage.mp/
possible, make issue for them

**Comment by plata** [comment]
> elaborate?! Me curious

Seems to be the successor of FiveM. See linked post.

**Comment by Kreyren** [comment]
From https://forum.fivem.net/t/the-onesync-eap-and-you/165931

>  ~~Run Windows. Linux servers may work, but there’s potential issues preventing players from seeing one another. This will be fixed at a future stage. Don’t ask for an ETA, you’ll risk losing access to the EAP.~~ Linux is fixed now. Rejoice!~ 

O.O

 @suomilanittaja Can you confirm that OneSync is supperior to FiveM ?

**Comment by suomilanittaja** [comment]
Idk what supperior men but text in fivem forum mean that linux server is fixed not client and they moderators said that fivem will not ever run on linux.https://forum.fivem.net/t/request-fivem-linux-client/182056/8

**Comment by ImperatorS79** [comment]
I will add the trustLevel like the environment in my PR #956.
